core/src/com/mygdx/minigolf/model/levels/LevelLoader.java

```diff
@@ -0,0 +1,81 @@
+package com.mygdx.minigolf.model.levels;
+
+import com.badlogic.ashley.core.Entity;
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.physics.box2d.CircleShape;
+import com.badlogic.gdx.physics.box2d.PolygonShape;
+import com.badlogic.gdx.physics.box2d.Shape;
+import com.mygdx.minigolf.controller.EntityFactory;
+import com.mygdx.minigolf.controller.EntityFactory.EntityType;
+import com.mygdx.minigolf.model.components.Physical;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.mygdx.minigolf.controller.EntityFactory.EntityType.SURFACE;
+
+
+public class LevelLoader {
+
+    static public List<Entity> loadLevel(String fileName) {
+        return loadLevel(CourseLoader.getCourse(fileName));
+    }
+
+    static public List<Entity> loadLevel(Course course) {
+        List<Entity> entities = new ArrayList<>();
+
+        // Add course itself as entity
+        Entity e = EntityFactory.get().createEntity(SURFACE);
+        Physical phys = e.getComponent(Physical.class);
+        phys.setPosition(new Vector2(0, 0));
+        PolygonShape shape = new PolygonShape();
+        shape.setAsBox(course.width, course.height);
+        phys.setShape(shape);
+        entities.add(e);
+
+        // Add all course elements as entities
+        for (CourseElement elem : course.getElements()) {
+            e = EntityFactory.get().createEntity(getEntityType(elem));
+
+            phys = e.getComponent(Physical.class);
```

Comment:
Kom over at denne måten å hente ut komponenter fra entities på er dyrere enn å lage ComponentMappers (O(log n) vs O(1), se [her](https://libgdx.badlogicgames.com/ci/ashley/docs/com/badlogic/ashley/core/Entity.html#getComponent-java.lang.Class-)). Vet ikke hvor pirkete vi skal være på det, siden vi ikke har så mange komponenter i hver entitet.

Reply:
Godt poeng! Hadde ikke tenkt på at det kunne være en forskjell på de to

Reply:
Tipper grunnen er componentMapper er en hash-struktur mens getComponent bruker en sortert liste eller et binærtré. 
Generelt sett vil hashstrukturer bruke mer minne, så det er kanskje ikke så velidg lurt å opprette de i hytt og pinne heller. Den beste løsningen vil kanskje være å lage globale componentmappers slik at vi kun trenger én for hver komponent istedenfor et helt sett for hvert objekt/statiske klasse som må gjøre en mapping.

Reply:
Et annet spm er spm er så klart hvilke konstante faktorer de to lookup-metodene har. Lookups i en hashtabell krever så klart at man utfører en hash-operasjon på objektet, gjør en tabell-lookup og så finner objektet. Lookups i lister/trær er litt mer direkte og jeg vil tro dataen vil ligge nærmere i minne, noe som vil være bra for caching. Men det er vanskelig å gjøre antagelser om slikt.

Reply:
Hmm, det er sant. Det høres fornuftig ut med globale mappers.